### ISSUE TITLE
Error when using the directive if the context path is not root

### ISSUE THREAD

**Issue by maeisabelle** [patrickmcelwee/ml-visjs-graph-ng#31]
I used the ml-visjs-graph directive in my application and it's working well when my app is in the root context path but when I tried to deploy my app in a path like http://localhost:8080/<contextpath>/, the call is pointing to http://localhost:8080/v1/resource instead of http://localhost:8080/<contextpath>/v1/resource. I think we need  to set the VisjsGraphService api to v1/resource instead of /v1/resource. In the current version, is there a way to do that? I can see that the service has setApi but I think it's not exposed in the directive.

**Comment by grtjn** [comment]
I think you can do as follows:
- In your application controller, inject `visjsGraphService`.
- Use `setApi('v1/resources')` on that
- Export the service to the scope or as a property of the ctrl
- add `graph-search="ctrl.service.graphSearch"` and `graph-expand="ctrl.service.graphExpand"` to the directive tag in your view html

That ought to do the trick..

**Comment by maeisabelle** [comment]
Hi @grtjn ,

setApi is not exposed because of the this.$get just limited the functions to search and expand. I'll try changing both functions instead. Thanks for the tip! :)

**Comment by grtjn** [comment]
Right, but it IS exposed. You can access it via the Provider that is automatically created for the service. You should be able to add this in your app.js (or similar):

  angular.modules('app').config(['visjsGraphServiceProvider', function(visjsGraphServiceProvider) {
    visjsGraphServiceProvider.setApi('v1/resources');
  }]);

**Comment by grtjn** [comment]
@maeisabelle Above was tested locally both by me and @patrickmcelwee. That should work.

I'd like to keep the ticket open though. Patrick and I both think it would be useful to expose that setApi on the service object itself too..

**Comment by maeisabelle** [comment]
Thanks @grtjn and @patrickmcelwee :) Very useful! Thanks for the prompt response.
